### ISSUE TITLE
`reportAny` on partial `Any`s

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#82]
```py
foo: list[Any] # should be an error
```

however there's tons of functions that take `Any` in the stdlib, eg.
```py
def cast(typ: type[_T], val: Any) -> _T: ...
```
these should not report an error. possible solutions:

- don't report `Any` errors on function calls where the argument takes `Any`, since there'll either be an error on the function definition anyway, or the function is third party/stdlib and therefore not within the user's control (and it usually means `object` in these cases anyway)
- remove all `Any`s from typeshed
- special-casing (horrible idea)

**Comment by DetachHead** [comment]
i think just reporting all explicit usages of `Any` will cover this case, and will match mypy's behavior.

reporting partial `Any` in the same way as partial unknown doesn't seem feasible because there's just way too much `Any` in the stdlib
